skimage/restoration/_nl_means_denoising.pyx

```diff
@@ -178,7 +177,7 @@ def _nl_means_denoising_2d(image, int s=7, int d=13, float h=0.1):
     cdef int row_start, row_end, col_start, col_end
     cdef int row_start_i, row_end_i, col_start_j, col_end_j
     cdef IMGDTYPE [::1] new_values = np.zeros(n_ch).astype(np.float32)
-    cdef IMGDTYPE [:, :, ::1] padded = np.ascontiguousarray(util.pad(image,
+    cdef IMGDTYPE [:, :, ::1] padded = np.ascontiguousarray(np.pad(image,
```

Comment:
Hiding functionality in different places does not endear an ecosystem to new users. We do image processing here; we should expose the ability to pad. Assuming users will find that feature in a different (even foundational) package is not intuitive. It's easy to forget this perspective when using all of these packages on a daily basis.

It's simple enough to add a line to the docstring saying it's a simple wrapper for `np.pad` while still grabbing it programmatically, but I am :-1: on removing this - now or in future. We may have to agree to disagree here.